lib/metrics.py

```diff
@@ -348,3 +348,41 @@ def _get_db_conn(self):
                 psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT
             )
         return self.db
+
+
+class FreeAppsMetricsEmitterThread(MetricsEmitterThread):
+    def run(self):
```

Comment:
What do you think of making this a template method in the base class? The structure is the same as in the `MetricsEmitterThread.run`, except for gathering stats. So by extracting the stats injection to its own method in the base class would allow to only override that and not having to copy the loop, sleeping and exception handling (which might cause problems in the future if we need to handle another exception or the current ones differently - we have to remember to change this in 2 places).

Reply:
As suggested, I have change MetricsEmitterThread to BaseMetricsEmitterThread, form which FreeAppsMetricsEmitterThread and PaidAppsMetricsEmitterThread inherit.